Milestone4/OfflineStudentManagementSystem/Control/services/staffServices.ts

```diff
@@ -0,0 +1,184 @@
+  return new Promise((res) => {
+    readLine.question(prompt, (answer: string) => {
+      res(answer);
+    });
+  });
+};
+
+export const staffLogin = async () => {
+  try {
+    const type = 'teacher';
+    const name = await questionAsync('Enter staff name: ');
+    const password = await questionAsync('Enter staff password: ');
+    validateStaffLogin(name, password);
+    Teacher.instanceOfTeacher.isActive = true;
+    if (Teacher.instanceOfTeacher.isActive) {
+      staffOperations();
+    } else {
+      throw new Error('Not logged In')
+    }
+    return staffMenu();
+  } catch (err) {
+    let message;
+    if (err instanceof Error) {
+      message = err.message;
+    } else {
+      message = String(message);
+    }
+    reportError({ message })
+  }
+};
+
+export const addStudentDetails = async () => {
+  try {
+    if (Teacher.instanceOfTeacher.isActive) {
+      const getRollNo = Number(
+        await questionAsync('Enter Roll No to add details : ')
+      );
+      const isExist = checkIsExist(getRollNo);
+      if (await isExist) {
+        const mark1 = Number(await questionAsync('Enter mark 1 : '));
+        const mark2 = Number(await questionAsync('Enter mark 2 : '));
+        const mark3 = Number(await questionAsync('Enter mark 3 : '));
+        const total = mark1 + mark2 + mark3;
+        const grade: string = calculateGrade(total);
+        const studentDetails = {
+          mark1,
+          mark2,
+          mark3,
+          total,
+          grade,
+        };
+        instantStudent.updateStudentSetter(studentDetails, getRollNo);
+        console.log(`\tDetails Added Succesfully\t`);
+        return staffMenu();
+      } else {
+        console.log('Roll Number not Found');
+        return staffMenu();
+      }
+    } else {
+      console.log(`\tOnly Staff can Add the details\t`);
+      return menu();
+    }
+  } catch (err) {
+    console.log(err);
+  }
+};
+
+export const calculateGrade = (total: number): string => {
+  if (total >= 285) {
+    return 'O';
+  } else if (total >= 250 && total <= 285) {
+    return 'A';
+  } else if (total >= 200 && total <= 250) {
+    return 'B';
+  } else if (total >= 150 && total <= 100) {
+    return 'C';
+  } else {
+    return 'Fail';
+  }
+};
+
+export const updateStudentDetails = async () => {
+  try {
+    const getRollNo = Number(
+      await questionAsync('Enter student Roll no to update : ')
+    );
+    const isExist = checkIsExist(getRollNo);
+    if (await isExist) {
+      const mark1 = Number(await questionAsync('Enter mark 1 : '));
+      const mark2 = Number(await questionAsync('Enter mark 2 : '));
+      const mark3 = Number(await questionAsync('Enter mark 3 : '));
+      const total = mark1 + mark2 + mark3;
+      const grade: string = calculateGrade(total);
+      const studentDetails = {
+        mark1,
+        mark2,
+        mark3,
+        total,
+        grade,
+      };
+      instantStudent.updateStudentSetter(studentDetails, getRollNo);
```

Comment:
can we change this varable name `instantStudent` to proper name

Reply:
resolved